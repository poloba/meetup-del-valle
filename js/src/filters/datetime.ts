import { DateTimeFormatOptions } from "vue-i18n";

function parseDateTime(value: string): Date {
  return new Date(value);
}

function formatDateString(value: string): string {
  return parseDateTime(value).toLocaleString(undefined, {
    weekday: "long",
    year: "numeric",
    month: "long",
    day: "numeric",
  });
}

function formatTimeString(value: string): string {
  return parseDateTime(value).toLocaleTimeString(undefined, {
    hour: "numeric",
    minute: "numeric",
  });
}

function formatDateTimeString(value: string, showTime = true): string {
  const options: DateTimeFormatOptions = {
    weekday: undefined,
    year: "numeric",
    month: "long",
    day: "numeric",
    hour: undefined,
    minute: undefined,
  };
  if (showTime) {
    options.weekday = "long";
    options.hour = "numeric";
    options.minute = "numeric";
  }
  const format = new Intl.DateTimeFormat(undefined, options);
  return format.format(parseDateTime(value));
}

export { formatDateString, formatTimeString, formatDateTimeString };
