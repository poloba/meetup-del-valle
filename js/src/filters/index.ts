import nl2br from "@/filters/utils";
import {
  formatDateString,
  formatTimeString,
  formatDateTimeString,
} from "./datetime";

export default {
  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
  install(vue: any): void {
    vue.filter("formatDateString", formatDateString);
    vue.filter("formatTimeString", formatTimeString);
    vue.filter("formatDateTimeString", formatDateTimeString);
    vue.filter("nl2br", nl2br);
  },
};
