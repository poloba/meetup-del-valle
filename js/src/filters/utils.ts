/**
 * New Line to <br>
 *
 * @param {string} str Input text
 * @return {string} Filtered text
 */
export default function nl2br(str: string): string {
  return `${str}`.replace(/([^>\r\n]?)(\r\n|\n\r|\r|\n)/g, "$1<br>");
}
