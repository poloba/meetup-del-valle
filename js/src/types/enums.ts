/* eslint-disable no-unused-vars */
export enum InstanceTermsType {
  DEFAULT = "DEFAULT",
  URL = "URL",
  CUSTOM = "CUSTOM",
}

export enum InstancePrivacyType {
  DEFAULT = "DEFAULT",
  URL = "URL",
  CUSTOM = "CUSTOM",
}

export enum ICurrentUserRole {
  USER = "USER",
  MODERATOR = "MODERATOR",
  ADMINISTRATOR = "ADMINISTRATOR",
}

export enum INotificationPendingEnum {
  NONE = "NONE",
  DIRECT = "DIRECT",
  ONE_DAY = "ONE_DAY",
  ONE_HOUR = "ONE_HOUR",
}

export enum IAuthProvider {
  LDAP = "ldap",
  GOOGLE = "google",
  DISCORD = "discord",
  GITHUB = "github",
  KEYCLOAK = "keycloak",
  FACEBOOK = "facebook",
  GITLAB = "gitlab",
  TWITTER = "twitter",
}

export enum ErrorCode {
  UNKNOWN = "unknown",
  REGISTRATION_CLOSED = "registration_closed",
}

export enum CommentModeration {
  ALLOW_ALL = "ALLOW_ALL",
  MODERATED = "MODERATED",
  CLOSED = "CLOSED",
}

export enum EventStatus {
  TENTATIVE = "TENTATIVE",
  CONFIRMED = "CONFIRMED",
  CANCELLED = "CANCELLED",
}

export enum EventVisibility {
  PUBLIC = "PUBLIC",
  UNLISTED = "UNLISTED",
  RESTRICTED = "RESTRICTED",
  PRIVATE = "PRIVATE",
}

export enum EventJoinOptions {
  FREE = "FREE",
  RESTRICTED = "RESTRICTED",
  INVITE = "INVITE",
}

export enum EventVisibilityJoinOptions {
  PUBLIC = "PUBLIC",
  LINK = "LINK",
  LIMITED = "LIMITED",
}

export enum Category {
  BUSINESS = "business",
  CONFERENCE = "conference",
  BIRTHDAY = "birthday",
  DEMONSTRATION = "demonstration",
  MEETING = "meeting",
}

export enum LoginErrorCode {
  NEED_TO_LOGIN = "need_to_login",
}

export enum LoginError {
  USER_NOT_CONFIRMED = "User account not confirmed",
  USER_DOES_NOT_EXIST = "No user with this email was found",
  USER_EMAIL_PASSWORD_INVALID = "Impossible to authenticate, either your email or password are invalid.",
  LOGIN_PROVIDER_ERROR = "Error with Login Provider",
  LOGIN_PROVIDER_NOT_FOUND = "Login Provider not found",
  USER_DISABLED = "This user has been disabled",
}

export enum ResetError {
  USER_IMPOSSIBLE_TO_RESET = "This user can't reset their password",
}

export enum ParticipantRole {
  NOT_APPROVED = "NOT_APPROVED",
  NOT_CONFIRMED = "NOT_CONFIRMED",
  REJECTED = "REJECTED",
  PARTICIPANT = "PARTICIPANT",
  MODERATOR = "MODERATOR",
  ADMINISTRATOR = "ADMINISTRATOR",
  CREATOR = "CREATOR",
}

export enum PostVisibility {
  PUBLIC = "PUBLIC",
  UNLISTED = "UNLISTED",
  RESTRICTED = "RESTRICTED",
  PRIVATE = "PRIVATE",
}

export enum ReportStatusEnum {
  OPEN = "OPEN",
  CLOSED = "CLOSED",
  RESOLVED = "RESOLVED",
}

export enum ActionLogAction {
  NOTE_CREATION = "NOTE_CREATION",
  NOTE_DELETION = "NOTE_DELETION",
  REPORT_UPDATE_CLOSED = "REPORT_UPDATE_CLOSED",
  REPORT_UPDATE_OPENED = "REPORT_UPDATE_OPENED",
  REPORT_UPDATE_RESOLVED = "REPORT_UPDATE_RESOLVED",
  EVENT_DELETION = "EVENT_DELETION",
  COMMENT_DELETION = "COMMENT_DELETION",
  ACTOR_SUSPENSION = "ACTOR_SUSPENSION",
  ACTOR_UNSUSPENSION = "ACTOR_UNSUSPENSION",
  USER_DELETION = "USER_DELETION",
}

export enum SearchTabs {
  EVENTS = 0,
  GROUPS = 1,
}

export enum ActorType {
  PERSON = "PERSON",
  APPLICATION = "APPLICATION",
  GROUP = "GROUP",
  ORGANISATION = "ORGANISATION",
  SERVICE = "SERVICE",
}

export enum MemberRole {
  NOT_APPROVED = "NOT_APPROVED",
  INVITED = "INVITED",
  MEMBER = "MEMBER",
  MODERATOR = "MODERATOR",
  ADMINISTRATOR = "ADMINISTRATOR",
  CREATOR = "CREATOR",
  REJECTED = "REJECTED",
}

export enum Openness {
  INVITE_ONLY = "INVITE_ONLY",
  MODERATED = "MODERATED",
  OPEN = "OPEN",
}

export enum RoutingType {
  OPENSTREETMAP = "OPENSTREETMAP",
  GOOGLE_MAPS = "GOOGLE_MAPS",
}

export enum RoutingTransportationType {
  FOOT = "FOOT",
  BIKE = "BIKE",
  TRANSIT = "TRANSIT",
  CAR = "CAR",
}

export enum GroupVisibility {
  PUBLIC = "PUBLIC",
  UNLISTED = "UNLISTED",
  PRIVATE = "PRIVATE",
}

export enum AddressSearchType {
  ADMINISTRATIVE = "ADMINISTRATIVE",
}
