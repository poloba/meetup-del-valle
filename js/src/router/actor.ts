import { RouteConfig } from "vue-router";
import { EsModuleComponent } from "vue/types/options";

export enum ActorRouteName {
  GROUP = "Group",
  CREATE_GROUP = "CreateGroup",
  PROFILE = "Profile",
  MY_GROUPS = "MY_GROUPS",
}

export const actorRoutes: RouteConfig[] = [
  {
    path: "/groups/create",
    name: ActorRouteName.CREATE_GROUP,
    component: (): Promise<EsModuleComponent> =>
      import(/* webpackChunkName: "CreateGroup" */ "@/views/Group/Create.vue"),
    meta: { requiredAuth: true },
  },
  {
    path: "/@:preferredUsername",
    name: ActorRouteName.GROUP,
    component: (): Promise<EsModuleComponent> =>
      import(/* webpackChunkName: "Group" */ "@/views/Group/Group.vue"),
    props: true,
    meta: { requiredAuth: false },
  },
  {
    path: "/groups/me",
    name: ActorRouteName.MY_GROUPS,
    component: (): Promise<EsModuleComponent> =>
      import(/* webpackChunkName: "MyGroups" */ "@/views/Group/MyGroups.vue"),
    meta: { requiredAuth: true },
  },
];
