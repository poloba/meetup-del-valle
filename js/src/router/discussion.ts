import { RouteConfig } from "vue-router";
import { EsModuleComponent } from "vue/types/options";

export enum DiscussionRouteName {
  DISCUSSION_LIST = "DISCUSSION_LIST",
  CREATE_DISCUSSION = "CREATE_DISCUSSION",
  DISCUSSION = "DISCUSSION",
}

export const discussionRoutes: RouteConfig[] = [
  {
    path: "/@:preferredUsername/discussions",
    name: DiscussionRouteName.DISCUSSION_LIST,
    component: (): Promise<EsModuleComponent> =>
      import(
        /* webpackChunkName: "DiscussionsList" */ "@/views/Discussions/DiscussionsList.vue"
      ),
    props: true,
    meta: { requiredAuth: true },
  },
  {
    path: "/@:preferredUsername/discussions/new",
    name: DiscussionRouteName.CREATE_DISCUSSION,
    component: (): Promise<EsModuleComponent> =>
      import(
        /* webpackChunkName: "CreateDiscussion" */ "@/views/Discussions/Create.vue"
      ),
    props: true,
    meta: { requiredAuth: true },
  },
  {
    path: "/@:preferredUsername/c/:slug/:comment_id?",
    name: DiscussionRouteName.DISCUSSION,
    component: (): Promise<EsModuleComponent> =>
      import(
        /* webpackChunkName: "Discussion" */ "@/views/Discussions/Discussion.vue"
      ),
    props: true,
    meta: { requiredAuth: true },
  },
];
