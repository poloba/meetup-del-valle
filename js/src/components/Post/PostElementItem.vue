<template>
  <router-link
    class="post-minimalist-card-wrapper"
    :to="{ name: RouteName.POST, params: { slug: post.slug } }"
  >
    <div class="title-info-wrapper">
      <div class="media">
        <div class="media-left">
          <figure class="image is-96x96" v-if="post.picture">
            <img :src="post.picture.url" alt="" />
          </figure>
          <b-icon v-else size="is-large" icon="post" />
        </div>
        <div class="media-content">
          <p class="post-minimalist-title">{{ post.title }}</p>
          <div class="metadata">
            <b-tag type="is-warning" size="is-small" v-if="post.draft">{{
              $t("Draft")
            }}</b-tag>
            <small
              v-if="
                post.visibility === PostVisibility.PUBLIC &&
                isCurrentActorMember
              "
              class="has-text-grey"
            >
              <b-icon icon="earth" size="is-small" />{{ $t("Public") }}</small
            >
            <small
              v-else-if="post.visibility === PostVisibility.UNLISTED"
              class="has-text-grey"
            >
              <b-icon icon="link" size="is-small" />{{
                $t("Accessible through link")
              }}</small
            >
            <small
              v-else-if="post.visibility === PostVisibility.PRIVATE"
              class="has-text-grey"
            >
              <b-icon icon="lock" size="is-small" />{{
                $t("Accessible only to members", {
                  group: post.attributedTo.name,
                })
              }}</small
            >
            <small class="has-text-grey">{{
              $options.filters.formatDateTimeString(
                new Date(post.insertedAt),
                false
              )
            }}</small>
            <small class="has-text-grey" v-if="isCurrentActorMember">{{
              $t("Created by {username}", {
                username: `@${usernameWithDomain(post.author)}`,
              })
            }}</small>
          </div>
        </div>
      </div>
    </div>
  </router-link>
</template>
<script lang="ts">
import { usernameWithDomain } from "@/types/actor";
import { PostVisibility } from "@/types/enums";
import { Component, Prop, Vue } from "vue-property-decorator";
import RouteName from "../../router/name";
import { IPost } from "../../types/post.model";

@Component
export default class PostElementItem extends Vue {
  @Prop({ required: true, type: Object }) post!: IPost;

  @Prop({ required: false, type: Boolean, default: false })
  isCurrentActorMember!: boolean;

  RouteName = RouteName;

  usernameWithDomain = usernameWithDomain;

  PostVisibility = PostVisibility;
}
</script>
<style lang="scss" scoped>
.post-minimalist-card-wrapper {
  text-decoration: none;
  display: flex;
  width: 100%;
  color: initial;
  border-bottom: 1px solid #e9e9e9;
  align-items: center;

  .title-info-wrapper {
    flex: 2;

    .post-minimalist-title {
      color: #3c376e;
      font-family: "Liberation Sans", "Helvetica Neue", Roboto, Helvetica, Arial,
        serif;
      font-size: 1rem;
      font-weight: 700;
      overflow: hidden;
      display: -webkit-box;
      -webkit-box-orient: vertical;
      -webkit-line-clamp: 2;
    }

    .media .media-left {
      & > span.icon {
        height: 96px;
        width: 96px;
      }
      & > figure.image > img {
        object-fit: cover;
        height: 100%;
        object-position: center;
        width: 100%;
      }
    }

    .metadata {
      & > span.tag {
        margin-right: 5px;
      }
      & > small:not(:last-child):after {
        content: "Â·";
        padding: 0 5px;
      }
    }
  }
}
</style>
