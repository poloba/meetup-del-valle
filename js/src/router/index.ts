import Vue from "vue";
import Router, { Route } from "vue-router";
import VueScrollTo from "vue-scrollto";
import { PositionResult } from "vue-router/types/router.d";
import { EsModuleComponent } from "vue/types/options";
import Home from "../views/Home.vue";
import { eventRoutes } from "./event";
import { actorRoutes } from "./actor";
import { errorRoutes } from "./error";
import { authGuardIfNeeded } from "./guards/auth-guard";
import { settingsRoutes } from "./settings";
import { groupsRoutes } from "./groups";
import { discussionRoutes } from "./discussion";
import { userRoutes } from "./user";
import RouteName from "./name";

Vue.use(Router);

function scrollBehavior(
  to: Route,
  from: Route,
  savedPosition: any
): PositionResult | undefined | null {
  if (to.hash) {
    VueScrollTo.scrollTo(to.hash, 700);
    return {
      selector: to.hash,
      offset: { x: 0, y: 10 },
    };
  }
  if (savedPosition) {
    return savedPosition;
  }

  return { x: 0, y: 0 };
}

export const routes = [
  ...userRoutes,
  ...eventRoutes,
  ...settingsRoutes,
  ...actorRoutes,
  ...groupsRoutes,
  ...discussionRoutes,
  ...errorRoutes,
  {
    path: "/search",
    name: RouteName.SEARCH,
    component: (): Promise<EsModuleComponent> =>
      import(/* webpackChunkName: "search" */ "../views/Search.vue"),
    props: true,
    meta: { requiredAuth: false },
  },
  {
    path: "/",
    name: RouteName.HOME,
    component: Home,
    meta: { requiredAuth: false },
  },
  {
    path: "/about",
    name: RouteName.ABOUT,
    component: (): Promise<EsModuleComponent> =>
      import(/* webpackChunkName: "about" */ "@/views/About.vue"),
    meta: { requiredAuth: false },
    redirect: { name: RouteName.ABOUT_INSTANCE },
    children: [
      {
        path: "instance",
        name: RouteName.ABOUT_INSTANCE,
        component: (): Promise<EsModuleComponent> =>
          import(
            /* webpackChunkName: "about" */ "@/views/About/AboutInstance.vue"
          ),
      },
      {
        path: "/terms",
        name: RouteName.TERMS,
        component: (): Promise<EsModuleComponent> =>
          import(/* webpackChunkName: "cookies" */ "@/views/About/Terms.vue"),
        meta: { requiredAuth: false },
      },
      {
        path: "/privacy",
        name: RouteName.PRIVACY,
        component: (): Promise<EsModuleComponent> =>
          import(/* webpackChunkName: "cookies" */ "@/views/About/Privacy.vue"),
        meta: { requiredAuth: false },
      },
      {
        path: "/rules",
        name: RouteName.RULES,
        component: (): Promise<EsModuleComponent> =>
          import(/* webpackChunkName: "cookies" */ "@/views/About/Rules.vue"),
        meta: { requiredAuth: false },
      },
      {
        path: "/glossary",
        name: RouteName.GLOSSARY,
        component: (): Promise<EsModuleComponent> =>
          import(
            /* webpackChunkName: "cookies" */ "@/views/About/Glossary.vue"
          ),
        meta: { requiredAuth: false },
      },
    ],
  },
  {
    path: "/interact",
    name: RouteName.INTERACT,
    component: (): Promise<EsModuleComponent> =>
      import(/* webpackChunkName: "cookies" */ "@/views/Interact.vue"),
    meta: { requiredAuth: false },
  },
  {
    path: "/auth/:provider/callback",
    name: "auth-callback",
    component: (): Promise<EsModuleComponent> =>
      import(
        /* webpackChunkName: "ProviderValidation" */ "@/views/User/ProviderValidation.vue"
      ),
  },
  {
    path: "/welcome/:step?",
    name: RouteName.WELCOME_SCREEN,
    component: (): Promise<EsModuleComponent> =>
      import(
        /* webpackChunkName: "WelcomeScreen" */ "@/views/User/SettingsOnboard.vue"
      ),
    meta: { requiredAuth: true },
    props: (route: Route): Record<string, unknown> => {
      const step = Number.parseInt(route.params.step, 10);
      if (Number.isNaN(step)) {
        return { step: 1 };
      }
      return { step };
    },
  },
  {
    path: "/404",
    name: RouteName.PAGE_NOT_FOUND,
    component: (): Promise<EsModuleComponent> =>
      import(/* webpackChunkName: "search" */ "../views/PageNotFound.vue"),
    meta: { requiredAuth: false },
  },
  {
    path: "*",
    redirect: { name: RouteName.PAGE_NOT_FOUND },
  },
];

const router = new Router({
  scrollBehavior,
  mode: "history",
  base: "/",
  routes,
});

router.beforeEach(authGuardIfNeeded);
router.afterEach(() => {
  if (router.app.$children[0]) {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    router.app.$children[0].error = null;
  }
});

export default router;
