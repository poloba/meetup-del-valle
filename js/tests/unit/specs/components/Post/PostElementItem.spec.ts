import { config, createLocalVue, mount } from "@vue/test-utils";
import PostElementItem from "@/components/Post/PostElementItem.vue";
import { formatDateTimeString } from "@/filters/datetime";
import Buefy from "buefy";
import VueRouter from "vue-router";
import { routes } from "@/router";
import { PostVisibility } from "@/types/enums";

const localVue = createLocalVue();
localVue.use(Buefy);
localVue.use(VueRouter);
const router = new VueRouter({ routes, mode: "history" });
localVue.filter("formatDateTimeString", formatDateTimeString);
config.mocks.$t = (key: string): string => key;

const postData = {
  id: "1",
  slug: "my-blog-post-some-uuid",
  title: "My Blog Post",
  body: "My content",
  insertedAt: "2020-12-02T09:01:20.873Z",
  visibility: PostVisibility.PUBLIC,
  author: {
    preferredUsername: "author",
    domain: "remote-domain.tld",
    name: "Author",
  },
  attributedTo: {
    preferredUsername: "my-awesome-group",
    domain: null,
    name: "My Awesome Group",
  },
};

const generateWrapper = (
  customPostData: Record<string, unknown> = {},
  isCurrentActorMember = false
) => {
  return mount(PostElementItem, {
    localVue,
    router,
    propsData: {
      post: { ...postData, ...customPostData },
      isCurrentActorMember,
    },
  });
};

describe("PostElementItem", () => {
  it("renders post with basic informations", () => {
    const wrapper = generateWrapper();
    expect(wrapper.html()).toMatchSnapshot();

    expect(
      wrapper.find("a.post-minimalist-card-wrapper").attributes("href")
    ).toBe(`/p/${postData.slug}`);

    expect(wrapper.find(".post-minimalist-title").text()).toContain(
      postData.title
    );
    expect(wrapper.find(".metadata").text()).toContain(
      formatDateTimeString(postData.insertedAt, false)
    );

    expect(wrapper.find(".metadata small").text()).not.toContain("Public");
  });

  it("shows the author if actor is a group member", () => {
    const wrapper = generateWrapper({}, true);
    expect(wrapper.html()).toMatchSnapshot();

    expect(wrapper.find(".metadata").text()).toContain(`Created by {username}`);
  });

  it("shows the draft tag if post is a draft", () => {
    const wrapper = generateWrapper({ draft: true });
    expect(wrapper.html()).toMatchSnapshot();

    expect(wrapper.findComponent({ name: "b-tag" }).exists()).toBe(true);
  });

  it("tells if the post is public when the actor is a group member", () => {
    const wrapper = generateWrapper({}, true);
    expect(wrapper.html()).toMatchSnapshot();

    expect(wrapper.find(".metadata small").text()).toContain("Public");
  });

  it("tells if the post is accessible only through link", () => {
    const wrapper = generateWrapper({ visibility: PostVisibility.UNLISTED });
    expect(wrapper.html()).toMatchSnapshot();

    expect(wrapper.find(".metadata small").text()).toContain(
      "Accessible through link"
    );
  });

  it("tells if the post is accessible only to members", () => {
    const wrapper = generateWrapper({ visibility: PostVisibility.PRIVATE });
    expect(wrapper.html()).toMatchSnapshot();

    expect(wrapper.find(".metadata small").text()).toContain(
      "Accessible only to members"
    );
  });
});
