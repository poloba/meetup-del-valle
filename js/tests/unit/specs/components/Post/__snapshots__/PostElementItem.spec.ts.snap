// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostElementItem renders post with basic informations 1`] = `
<a href="/p/my-blog-post-some-uuid" class="post-minimalist-card-wrapper">
  <div class="title-info-wrapper">
    <div class="media">
      <div class="media-left"><span class="icon is-large"><i class="mdi mdi-post mdi-48px"></i></span></div>
      <div class="media-content">
        <p class="post-minimalist-title">My Blog Post</p>
        <div class="metadata">
          <!---->
          <!----> <small class="has-text-grey">December 2, 2020</small>
          <!---->
        </div>
      </div>
    </div>
  </div>
</a>
`;

exports[`PostElementItem shows the author if actor is a group member 1`] = `
<a href="/p/my-blog-post-some-uuid" class="post-minimalist-card-wrapper">
  <div class="title-info-wrapper">
    <div class="media">
      <div class="media-left"><span class="icon is-large"><i class="mdi mdi-post mdi-48px"></i></span></div>
      <div class="media-content">
        <p class="post-minimalist-title">My Blog Post</p>
        <div class="metadata">
          <!----> <small class="has-text-grey"><span class="icon is-small"><i class="mdi mdi-earth"></i></span>Public</small> <small class="has-text-grey">December 2, 2020</small> <small class="has-text-grey">Created by {username}</small>
        </div>
      </div>
    </div>
  </div>
</a>
`;

exports[`PostElementItem shows the draft tag if post is a draft 1`] = `
<a href="/p/my-blog-post-some-uuid" class="post-minimalist-card-wrapper">
  <div class="title-info-wrapper">
    <div class="media">
      <div class="media-left"><span class="icon is-large"><i class="mdi mdi-post mdi-48px"></i></span></div>
      <div class="media-content">
        <p class="post-minimalist-title">My Blog Post</p>
        <div class="metadata"><span class="tag is-warning is-small"><span class="">Draft</span>
          <!----></span>
          <!----> <small class="has-text-grey">December 2, 2020</small>
          <!---->
        </div>
      </div>
    </div>
  </div>
</a>
`;

exports[`PostElementItem tells if the post is accessible only through link 1`] = `
<a href="/p/my-blog-post-some-uuid" class="post-minimalist-card-wrapper">
  <div class="title-info-wrapper">
    <div class="media">
      <div class="media-left"><span class="icon is-large"><i class="mdi mdi-post mdi-48px"></i></span></div>
      <div class="media-content">
        <p class="post-minimalist-title">My Blog Post</p>
        <div class="metadata">
          <!----> <small class="has-text-grey"><span class="icon is-small"><i class="mdi mdi-link"></i></span>Accessible through link</small> <small class="has-text-grey">December 2, 2020</small>
          <!---->
        </div>
      </div>
    </div>
  </div>
</a>
`;

exports[`PostElementItem tells if the post is accessible only to members 1`] = `
<a href="/p/my-blog-post-some-uuid" class="post-minimalist-card-wrapper">
  <div class="title-info-wrapper">
    <div class="media">
      <div class="media-left"><span class="icon is-large"><i class="mdi mdi-post mdi-48px"></i></span></div>
      <div class="media-content">
        <p class="post-minimalist-title">My Blog Post</p>
        <div class="metadata">
          <!----> <small class="has-text-grey"><span class="icon is-small"><i class="mdi mdi-lock"></i></span>Accessible only to members</small> <small class="has-text-grey">December 2, 2020</small>
          <!---->
        </div>
      </div>
    </div>
  </div>
</a>
`;

exports[`PostElementItem tells if the post is public when the actor is a group member 1`] = `
<a href="/p/my-blog-post-some-uuid" class="post-minimalist-card-wrapper">
  <div class="title-info-wrapper">
    <div class="media">
      <div class="media-left"><span class="icon is-large"><i class="mdi mdi-post mdi-48px"></i></span></div>
      <div class="media-content">
        <p class="post-minimalist-title">My Blog Post</p>
        <div class="metadata">
          <!----> <small class="has-text-grey"><span class="icon is-small"><i class="mdi mdi-earth"></i></span>Public</small> <small class="has-text-grey">December 2, 2020</small> <small class="has-text-grey">Created by {username}</small>
        </div>
      </div>
    </div>
  </div>
</a>
`;
